### ISSUE TITLE
ESP32 I2S linker error

### ISSUE THREAD

**Issue by Jueff** [FastLED/FastLED#1810]
Compiling the Blink sketch with FASTLED_ESP32_I2S defined fails with a lot of undefined reference.

Board: ESP32DevBoard
FastLED: 3.9.7  (when switching back to 3.6.0 problem is gone)
esp32:esp32: Espressif 1.0.4

Code
```
#define FASTLED_ESP32_I2S
#include <FastLED.h>
...
...
```

Build output
```
Archiving built core (caching) in: C:\Users\Tester\AppData\Local\arduino\cores\1420933ad65eba4cac1859e0a4bb7280\core.a
Linking everything together...
"C:\\Users\\....\\AppData\\Local\\Arduino15\\packages\\esp32\\tools\\xtensa-esp32-elf-gcc\\1.22.0-80-g6c4433a-5.2.0/bin/xtensa-esp32-elf-gcc" -nostdlib "-LC:\\Users\\....\\AppData\\Local\\Arduino15\\packages\\esp32\\hardware\\esp32\\1.0.4/tools/sdk/lib" "-LC:\\Users\\....\\AppData\\Local\\Arduino15\\packages\\esp32\\hardware\\esp32\\1.0.4/tools/sdk/ld" -T esp32_out.ld -T esp32.common.ld -T esp32.rom.ld -T esp32.peripherals.ld -T esp32.rom.libgcc.ld -T esp32.rom.spiram_incompatible_fns.ld -u ld_include_panic_highint_hdl -u call_user_start_cpu0 -Wl,--gc-sections -Wl,-static -Wl,--undefined=uxTopUsedPriority -u __cxa_guard_dummy -u __cxx_fatal_exception -Wl,--start-group "C:\\Users\\....\\AppData\\Local\\arduino\\sketches\\4EB0E45605664DFC0D280DCBCBFD79BC\\sketch\\Blink.ino.cpp.o" "C:\\Users\\....\\AppData\\Local\\arduino\\sketches\\4EB0E45605664DFC0D280DCBCBFD79BC\\libraries\\FastLED\\FastLED.cpp.o" "C:\\Users\\....\\AppData\\Local\\arduino\\sketches\\4EB0E45605664DFC0D280DCBCBFD79BC\\libraries\\FastLED\\bitswap.cpp.o" "C:\\Users\\....\\AppData\\Local\\arduino\\sketches\\4EB0E45605664DFC0D280DCBCBFD79BC\\libraries\\FastLED\\cled_controller.cpp.o" "C:\\Users\\....\\AppData\\Local\\arduino\\sketches\\4EB0E45605664DFC0D280DCBCBFD79BC\\libraries\\FastLED\\colorpalettes.cpp.o" "C:\\Users\\....\\AppData\\Local\\arduino\\sketches\\4EB0E45605664DFC0D280DCBCBFD79BC\\libraries\\FastLED\\colorutils.cpp.o" "C:\\Users\\....\\AppData\\Local\\arduino\\sketches\\4EB0E45605664DFC0D280DCBCBFD79BC\\libraries\\FastLED\\crgb.cpp.o" "C:\\Users\\....\\AppData\\Local\\arduino\\sketches\\4EB0E45605664DFC0D280DCBCBFD79BC\\libraries\\FastLED\\five_bit_hd_gamma.cpp.o" "C:\\Users\\....\\AppData\\Local\\arduino\\sketches\\4EB0E45605664DFC0D280DCBCBFD79BC\\libraries\\FastLED\\fl\\allocator.cpp.o" "C:\\Users\\....\\AppData\\Local\\arduino\\sketches\\4EB0E45605664DFC0D280DCBCBFD79BC\\libraries\\FastLED\\fl\\bytestreammemory.cpp.o" "C:\\Users\\....\\AppData\\Local\\arduino\\sketches\\4EB0E45605664DFC0D280DCBCBFD79BC\\libraries\\FastLED\\fl\\engine_events.cpp.o" "C:\\Users\\....\\AppData\\Local\\arduino\\sketches\\4EB0E45605664DFC0D280DCBCBFD79BC\\libraries\\FastLED\\fl\\file_system.cpp.o" "C:\\Users\\....\\AppData\\Local\\arduino\\sketches\\4EB0E45605664DFC0D280DCBCBFD79BC\\libraries\\FastLED\\fl\\json.cpp.o" "C:\\Users\\....\\AppData\\Local\\arduino\\sketches\\4EB0E45605664DFC0D280DCBCBFD79BC\\libraries\\FastLED\\fl\\ref.cpp.o" "C:\\Users\\....\\AppData\\Local\\arduino\\sketches\\4EB0E45605664DFC0D280DCBCBFD79BC\\libraries\\FastLED\\fl\\screenmap.cpp.o" "C:\\Users\\....\\AppData\\Local\\arduino\\sketches\\4EB0E45605664DFC0D280DCBCBFD79BC\\libraries\\FastLED\\fl\\str.cpp.o" "C:\\Users\\....\\AppData\\Local\\arduino\\sketches\\4EB0E45605664DFC0D280DCBCBFD79BC\\libraries\\FastLED\\fl\\stub_main.cpp.o" "C:\\Users\\....\\AppData\\Local\\arduino\\sketches\\4EB0E45605664DFC0D280DCBCBFD79BC\\libraries\\FastLED\\fl\\xmap.cpp.o" "C:\\Users\\....\\AppData\\Local\\arduino\\sketches\\4EB0E45605664DFC0D280DCBCBFD79BC\\libraries\\FastLED\\fl\\xymap.cpp.o" "C:\\Users\\....\\AppData\\Local\\arduino\\sketches\\4EB0E45605664DFC0D280DCBCBFD79BC\\libraries\\FastLED\\fx\\2d\\bilinear_expansion.cpp.o" "C:\\Users\\....\\AppData\\Local\\arduino\\sketches\\4EB0E45605664DFC0D280DCBCBFD79BC\\libraries\\FastLED\\fx\\2d\\noisepalette.cpp.o" "C:\\Users\\....\\AppData\\Local\\arduino\\sketches\\4EB0E45605664DFC0D280DCBCBFD79BC\\libraries\\FastLED\\fx\\2d\\scale_up.cpp.o" "C:\\Users\\....\\AppData\\Local\\arduino\\sketches\\4EB0E45605664DFC0D280DCBCBFD79BC\\libraries\\FastLED\\fx\\frame.cpp.o" "C:\\Users\\....\\AppData\\Local\\arduino\\sketches\\4EB0E45605664DFC0D280DCBCBFD79BC\\libraries\\FastLED\\fx\\fx_engine.cpp.o" "C:\\Users\\....\\AppData\\Local\\arduino\\sketches\\4EB0E45605664DFC0D280DCBCBFD79BC\\libraries\\FastLED\\fx\\time.cpp.o" "C:\\Users\\....\\AppData\\Local\\arduino\\sketches\\4EB0E45605664DFC0D280DCBCBFD79BC\\libraries\\FastLED\\fx\\video.cpp.o" "C:\\Users\\....\\AppData\\Local\\arduino\\sketches\\4EB0E45605664DFC0D280DCBCBFD79BC\\libraries\\FastLED\\fx\\video\\frame_interpolator.cpp.o" "C:\\Users\\....\\AppData\\Local\\arduino\\sketches\\4EB0E45605664DFC0D280DCBCBFD79BC\\libraries\\FastLED\\fx\\video\\frame_tracker.cpp.o" "C:\\Users\\....\\AppData\\Local\\arduino\\sketches\\4EB0E45605664DFC0D280DCBCBFD79BC\\libraries\\FastLED\\fx\\video\\pixel_stream.cpp.o" "C:\\Users\\....\\AppData\\Local\\arduino\\sketches\\4EB0E45605664DFC0D280DCBCBFD79BC\\libraries\\FastLED\\fx\\video\\video_impl.cpp.o" "C:\\Users\\....\\AppData\\Local\\arduino\\sketches\\4EB0E45605664DFC0D280DCBCBFD79BC\\libraries\\FastLED\\hsv2rgb.cpp.o" "C:\\Users\\....\\AppData\\Local\\arduino\\sketches\\4EB0E45605664DFC0D280DCBCBFD79BC\\libraries\\FastLED\\lib8tion.cpp.o" "C:\\Users\\....\\AppData\\Local\\arduino\\sketches\\4EB0E45605664DFC0D280DCBCBFD79BC\\libraries\\FastLED\\noise.cpp.o" "C:\\Users\\....\\AppData\\Local\\arduino\\sketches\\4EB0E45605664DFC0D280DCBCBFD79BC\\libraries\\FastLED\\platforms.cpp.o" "C:\\Users\\....\\AppData\\Local\\arduino\\sketches\\4EB0E45605664DFC0D280DCBCBFD79BC\\libraries\\FastLED\\platforms\\avr\\avr_millis_timer_source.cpp.o" "C:\\Users\\....\\AppData\\Local\\arduino\\sketches\\4EB0E45605664DFC0D280DCBCBFD79BC\\libraries\\FastLED\\platforms\\esp\\32\\experimental\\s3_clockless_and_clocked_driver.cpp.o" "C:\\Users\\....\\AppData\\Local\\arduino\\sketches\\4EB0E45605664DFC0D280DCBCBFD79BC\\libraries\\FastLED\\platforms\\esp\\32\\i2s.cpp.o" "C:\\Users\\....\\AppData\\Local\\arduino\\sketches\\4EB0E45605664DFC0D280DCBCBFD79BC\\libraries\\FastLED\\platforms\\esp\\32\\idf4_rmt.cpp.o" "C:\\Users\\....\\AppData\\Local\\arduino\\sketches\\4EB0E45605664DFC0D280DCBCBFD79BC\\libraries\\FastLED\\platforms\\esp\\32\\idf4_rmt_impl.cpp.o" "C:\\Users\\....\\AppData\\Local\\arduino\\sketches\\4EB0E45605664DFC0D280DCBCBFD79BC\\libraries\\FastLED\\platforms\\esp\\32\\idf5_rmt.cpp.o" "C:\\Users\\....\\AppData\\Local\\arduino\\sketches\\4EB0E45605664DFC0D280DCBCBFD79BC\\libraries\\FastLED\\platforms\\esp\\32\\led_strip\\configure_led.cpp.o" "C:\\Users\\....\\AppData\\Local\\arduino\\sketches\\4EB0E45605664DFC0D280DCBCBFD79BC\\libraries\\FastLED\\platforms\\esp\\32\\led_strip\\construct.cpp.o" "C:\\Users\\....\\AppData\\Local\\arduino\\sketches\\4EB0E45605664DFC0D280DCBCBFD79BC\\libraries\\FastLED\\platforms\\esp\\32\\led_strip\\led_strip_api.cpp.o" "C:\\Users\\....\\AppData\\Local\\arduino\\sketches\\4EB0E45605664DFC0D280DCBCBFD79BC\\libraries\\FastLED\\platforms\\esp\\32\\led_strip\\led_strip_rmt_dev.cpp.o" "C:\\Users\\....\\AppData\\Local\\arduino\\sketches\\4EB0E45605664DFC0D280DCBCBFD79BC\\libraries\\FastLED\\platforms\\esp\\32\\led_strip\\led_strip_rmt_encoder.cpp.o" "C:\\Users\\....\\AppData\\Local\\arduino\\sketches\\4EB0E45605664DFC0D280DCBCBFD79BC\\libraries\\FastLED\\platforms\\esp\\32\\led_strip\\rmt_strip.cpp.o" "C:\\Users\\....\\AppData\\Local\\arduino\\sketches\\4EB0E45605664DFC0D280DCBCBFD79BC\\libraries\\FastLED\\platforms\\esp\\32\\led_strip\\rmt_strip_group.cpp.o" "C:\\Users\\....\\AppData\\Local\\arduino\\sketches\\4EB0E45605664DFC0D280DCBCBFD79BC\\libraries\\FastLED\\platforms\\stub\\led_sysdefs_stub.cpp.o" "C:\\Users\\....\\AppData\\Local\\arduino\\sketches\\4EB0E45605664DFC0D280DCBCBFD79BC\\libraries\\FastLED\\platforms\\wasm\\active_strip_data.cpp.o" "C:\\Users\\....\\AppData\\Local\\arduino\\sketches\\4EB0E45605664DFC0D280DCBCBFD79BC\\libraries\\FastLED\\platforms\\wasm\\compiler\\Arduino.cpp.o" "C:\\Users\\....\\AppData\\Local\\arduino\\sketches\\4EB0E45605664DFC0D280DCBCBFD79BC\\libraries\\FastLED\\platforms\\wasm\\engine_listener.cpp.o" "C:\\Users\\....\\AppData\\Local\\arduino\\sketches\\4EB0E45605664DFC0D280DCBCBFD79BC\\libraries\\FastLED\\platforms\\wasm\\fs_wasm.cpp.o" "C:\\Users\\....\\AppData\\Local\\arduino\\sketches\\4EB0E45605664DFC0D280DCBCBFD79BC\\libraries\\FastLED\\platforms\\wasm\\js.cpp.o" "C:\\Users\\....\\AppData\\Local\\arduino\\sketches\\4EB0E45605664DFC0D280DCBCBFD79BC\\libraries\\FastLED\\platforms\\wasm\\timer.cpp.o" "C:\\Users\\....\\AppData\\Local\\arduino\\sketches\\4EB0E45605664DFC0D280DCBCBFD79BC\\libraries\\FastLED\\platforms\\wasm\\ui\\button.cpp.o" "C:\\Users\\....\\AppData\\Local\\arduino\\sketches\\4EB0E45605664DFC0D280DCBCBFD79BC\\libraries\\FastLED\\platforms\\wasm\\ui\\checkbox.cpp.o" "C:\\Users\\....\\AppData\\Local\\arduino\\sketches\\4EB0E45605664DFC0D280DCBCBFD79BC\\libraries\\FastLED\\platforms\\wasm\\ui\\description.cpp.o" "C:\\Users\\....\\AppData\\Local\\arduino\\sketches\\4EB0E45605664DFC0D280DCBCBFD79BC\\libraries\\FastLED\\platforms\\wasm\\ui\\slider.cpp.o" "C:\\Users\\....\\AppData\\Local\\arduino\\sketches\\4EB0E45605664DFC0D280DCBCBFD79BC\\libraries\\FastLED\\platforms\\wasm\\ui\\title.cpp.o" "C:\\Users\\....\\AppData\\Local\\arduino\\sketches\\4EB0E45605664DFC0D280DCBCBFD79BC\\libraries\\FastLED\\platforms\\wasm\\ui\\ui_internal.cpp.o" "C:\\Users\\....\\AppData\\Local\\arduino\\sketches\\4EB0E45605664DFC0D280DCBCBFD79BC\\libraries\\FastLED\\platforms\\wasm\\ui\\ui_manager.cpp.o" "C:\\Users\\....\\AppData\\Local\\arduino\\sketches\\4EB0E45605664DFC0D280DCBCBFD79BC\\libraries\\FastLED\\platforms\\wasm\\ui\\ui_number_field.cpp.o" "C:\\Users\\....\\AppData\\Local\\arduino\\sketches\\4EB0E45605664DFC0D280DCBCBFD79BC\\libraries\\FastLED\\power_mgt.cpp.o" "C:\\Users\\....\\AppData\\Local\\arduino\\sketches\\4EB0E45605664DFC0D280DCBCBFD79BC\\libraries\\FastLED\\rgbw.cpp.o" "C:\\Users\\....\\AppData\\Local\\arduino\\sketches\\4EB0E45605664DFC0D280DCBCBFD79BC\\libraries\\FastLED\\sensors\\digital_pin.cpp.o" "C:\\Users\\....\\AppData\\Local\\arduino\\sketches\\4EB0E45605664DFC0D280DCBCBFD79BC\\libraries\\FastLED\\sensors\\pir.cpp.o" "C:\\Users\\....\\AppData\\Local\\arduino\\sketches\\4EB0E45605664DFC0D280DCBCBFD79BC\\libraries\\FastLED\\simplex.cpp.o" "C:\\Users\\....\\AppData\\Local\\arduino\\sketches\\4EB0E45605664DFC0D280DCBCBFD79BC\\libraries\\FastLED\\third_party\\object_fled\\src\\OjectFLED.cpp.o" "C:\\Users\\....\\AppData\\Local\\arduino\\sketches\\4EB0E45605664DFC0D280DCBCBFD79BC\\libraries\\FastLED\\transpose8x1_noinline.cpp.o" "C:\\Users\\....\\AppData\\Local\\arduino\\sketches\\4EB0E45605664DFC0D280DCBCBFD79BC\\libraries\\FastLED\\wiring.cpp.o" "C:\\Users\\....\\AppData\\Local\\arduino\\sketches\\4EB0E45605664DFC0D280DCBCBFD79BC\\core\\core.a" -lgcc -lesp32 -lphy -lesp_http_client -lmbedtls -lrtc -lesp_http_server -lbtdm_app -lspiffs -lbootloader_support -lmdns -lnvs_flash -lfatfs -lpp -lnet80211 -ljsmn -lface_detection -llibsodium -lvfs -ldl_lib -llog -lfreertos -lcxx -lsmartconfig_ack -lxtensa-debug-module -lheap -ltcpip_adapter -lmqtt -lulp -lfd -lfb_gfx -lnghttp -lprotocomm -lsmartconfig -lm -lethernet -limage_util -lc_nano -lsoc -ltcp_transport -lc -lmicro-ecc -lface_recognition -ljson -lwpa_supplicant -lmesh -lesp_https_ota -lwpa2 -lexpat -llwip -lwear_levelling -lapp_update -ldriver -lbt -lespnow -lcoap -lasio -lnewlib -lconsole -lapp_trace -lesp32-camera -lhal -lprotobuf-c -lsdmmc -lcore -lpthread -lcoexist -lfreemodbus -lspi_flash -lesp-tls -lwpa -lwifi_provisioning -lwps -lesp_adc_cal -lesp_event -lopenssl -lesp_ringbuf -lfr -lstdc++ -Wl,--end-group -Wl,-EL -o "C:\\Users\\....\\AppData\\Local\\arduino\\sketches\\4EB0E45605664DFC0D280DCBCBFD79BC/Blink.ino.elf"
C:\Users\Tester\AppData\Local\arduino\sketches\4EB0E45605664DFC0D280DCBCBFD79BC\libraries\FastLED\transpose8x1_noinline.cpp.o: In function `transpose8x1_noinline(unsigned char*, unsigned char*)':
c:\Users\....\Documents\Arduino\libraries\FastLED\src/transpose8x1_noinline.cpp:8: multiple definition of `transpose8x1_noinline(unsigned char*, unsigned char*)'
C:\Users\Tester\AppData\Local\arduino\sketches\4EB0E45605664DFC0D280DCBCBFD79BC\libraries\FastLED\bitswap.cpp.o:c:\Users\....\Documents\Arduino\libraries\FastLED\src/bitswap.cpp:9: first defined here
C:\Users\Tester\AppData\Local\arduino\sketches\4EB0E45605664DFC0D280DCBCBFD79BC\sketch\Blink.ino.cpp.o:(.literal._ZN19ClocklessControllerILi16ELi60ELi150ELi90EL6EOrder66ELi0ELb0ELi5EE10fillBufferEv[ClocklessController<16, 60, 150, 90, (EOrder)66, 0, false, 5>::fillBuffer()]+0x0): undefined reference to `gCurBuffer'
C:\Users\Tester\AppData\Local\arduino\sketches\4EB0E45605664DFC0D280DCBCBFD79BC\sketch\Blink.ino.cpp.o:(.literal._ZN19ClocklessControllerILi16ELi60ELi150ELi90EL6EOrder66ELi0ELb0ELi5EE10fillBufferEv[ClocklessController<16, 60, 150, 90, (EOrder)66, 0, false, 5>::fillBuffer()]+0x4): undefined reference to `dmaBuffers'
C:\Users\Tester\AppData\Local\arduino\sketches\4EB0E45605664DFC0D280DCBCBFD79BC\sketch\Blink.ino.cpp.o:(.literal._ZN19ClocklessControllerILi16ELi60ELi150ELi90EL6EOrder66ELi0ELb0ELi5EE10fillBufferEv[ClocklessController<16, 60, 150, 90, (EOrder)66, 0, false, 5>::fillBuffer()]+0x14): undefined reference to `gPixelRow'
C:\Users\Tester\AppData\Local\arduino\sketches\4EB0E45605664DFC0D280DCBCBFD79BC\sketch\Blink.ino.cpp.o:(.literal._ZN19ClocklessControllerILi16ELi60ELi150ELi90EL6EOrder66ELi0ELb0ELi5EE10fillBufferEv[ClocklessController<16, 60, 150, 90, (EOrder)66, 0, false, 5>::fillBuffer()]+0x18): undefined reference to `gDoneFilling'
C:\Users\Tester\AppData\Local\arduino\sketches\4EB0E45605664DFC0D280DCBCBFD79BC\sketch\Blink.ino.cpp.o:(.literal._ZN19ClocklessControllerILi16ELi60ELi150ELi90EL6EOrder66ELi0ELb0ELi5EE10fillBufferEv[ClocklessController<16, 60, 150, 90, (EOrder)66, 0, false, 5>::fillBuffer()]+0x1c): undefined reference to `i2s_transpose_and_encode(int, unsigned int, unsigned int volatile*)'
C:\Users\Tester\AppData\Local\arduino\sketches\4EB0E45605664DFC0D280DCBCBFD79BC\sketch\Blink.ino.cpp.o:(.literal._ZN19ClocklessControllerILi16ELi60ELi150ELi90EL6EOrder66ELi0ELb0ELi5EE4initEv[ClocklessController<16, 60, 150, 90, (EOrder)66, 0, false, 5>::init()]+0x8): undefined reference to `i2s_is_initialized()'
C:\Users\Tester\AppData\Local\arduino\sketches\4EB0E45605664DFC0D280DCBCBFD79BC\sketch\Blink.ino.cpp.o:(.literal._ZN19ClocklessControllerILi16ELi60ELi150ELi90EL6EOrder66ELi0ELb0ELi5EE4initEv[ClocklessController<16, 60, 150, 90, (EOrder)66, 0, false, 5>::init()]+0xc): undefined reference to `i2s_define_bit_patterns(int, int, int)'
C:\Users\Tester\AppData\Local\arduino\sketches\4EB0E45605664DFC0D280DCBCBFD79BC\sketch\Blink.ino.cpp.o:(.literal._ZN19ClocklessControllerILi16ELi60ELi150ELi90EL6EOrder66ELi0ELb0ELi5EE4initEv[ClocklessController<16, 60, 150, 90, (EOrder)66, 0, false, 5>::init()]+0x10): undefined reference to `i2s_init(int)'
C:\Users\Tester\AppData\Local\arduino\sketches\4EB0E45605664DFC0D280DCBCBFD79BC\sketch\Blink.ino.cpp.o:(.literal._ZN19ClocklessControllerILi16ELi60ELi150ELi90EL6EOrder66ELi0ELb0ELi5EE4initEv[ClocklessController<16, 60, 150, 90, (EOrder)66, 0, false, 5>::init()]+0x14): undefined reference to `i2s_set_fill_buffer_callback(void (*)())'
C:\Users\Tester\AppData\Local\arduino\sketches\4EB0E45605664DFC0D280DCBCBFD79BC\sketch\Blink.ino.cpp.o:(.literal._ZN19ClocklessControllerILi16ELi60ELi150ELi90EL6EOrder66ELi0ELb0ELi5EE4initEv[ClocklessController<16, 60, 150, 90, (EOrder)66, 0, false, 5>::init()]+0x18): undefined reference to `i2s_setup_pin(int, int)'
C:\Users\Tester\AppData\Local\arduino\sketches\4EB0E45605664DFC0D280DCBCBFD79BC\sketch\Blink.ino.cpp.o:(.literal._ZN19ClocklessControllerILi16ELi60ELi150ELi90EL6EOrder66ELi0ELb0ELi5EE10showPixelsER15PixelControllerILS0_66ELi1ELj4294967295EE[ClocklessController<16, 60, 150, 90, (EOrder)66, 0, false, 5>::showPixels(PixelController<(EOrder)66, 1, 4294967295u>&)]+0x4): undefined reference to `i2s_begin()'
C:\Users\Tester\AppData\Local\arduino\sketches\4EB0E45605664DFC0D280DCBCBFD79BC\sketch\Blink.ino.cpp.o:(.literal._ZN19ClocklessControllerILi16ELi60ELi150ELi90EL6EOrder66ELi0ELb0ELi5EE10showPixelsER15PixelControllerILS0_66ELi1ELj4294967295EE[ClocklessController<16, 60, 150, 90, (EOrder)66, 0, false, 5>::showPixels(PixelController<(EOrder)66, 1, 4294967295u>&)]+0xc): undefined reference to `i2s_clear_dma_buffer(unsigned int*)'
C:\Users\Tester\AppData\Local\arduino\sketches\4EB0E45605664DFC0D280DCBCBFD79BC\sketch\Blink.ino.cpp.o:(.literal._ZN19ClocklessControllerILi16ELi60ELi150ELi90EL6EOrder66ELi0ELb0ELi5EE10showPixelsER15PixelControllerILS0_66ELi1ELj4294967295EE[ClocklessController<16, 60, 150, 90, (EOrder)66, 0, false, 5>::showPixels(PixelController<(EOrder)66, 1, 4294967295u>&)]+0x10): undefined reference to `i2s_start()'
C:\Users\Tester\AppData\Local\arduino\sketches\4EB0E45605664DFC0D280DCBCBFD79BC\sketch\Blink.ino.cpp.o:(.literal._ZN19ClocklessControllerILi16ELi60ELi150ELi90EL6EOrder66ELi0ELb0ELi5EE10showPixelsER15PixelControllerILS0_66ELi1ELj4294967295EE[ClocklessController<16, 60, 150, 90, (EOrder)66, 0, false, 5>::showPixels(PixelController<(EOrder)66, 1, 4294967295u>&)]+0x14): undefined reference to `i2s_wait()'
C:\Users\Tester\AppData\Local\arduino\sketches\4EB0E45605664DFC0D280DCBCBFD79BC\sketch\Blink.ino.cpp.o:(.literal._ZN19ClocklessControllerILi16ELi60ELi150ELi90EL6EOrder66ELi0ELb0ELi5EE10showPixelsER15PixelControllerILS0_66ELi1ELj4294967295EE[ClocklessController<16, 60, 150, 90, (EOrder)66, 0, false, 5>::showPixels(PixelController<(EOrder)66, 1, 4294967295u>&)]+0x18): undefined reference to `i2s_stop()'
C:\Users\Tester\AppData\Local\arduino\sketches\4EB0E45605664DFC0D280DCBCBFD79BC\sketch\Blink.ino.cpp.o: In function `ClocklessController<16, 60, 150, 90, (EOrder)66, 0, false, 5>::fillBuffer()':
c:\Users\....\Documents\Arduino\libraries\FastLED\src/platforms/esp/32/clockless_i2s_esp32.h:249: undefined reference to `i2s_transpose_and_encode(int, unsigned int, unsigned int volatile*)'
C:\Users\Tester\AppData\Local\arduino\sketches\4EB0E45605664DFC0D280DCBCBFD79BC\sketch\Blink.ino.cpp.o: In function `ClocklessController<16, 60, 150, 90, (EOrder)66, 0, false, 5>::init()':
c:\Users\....\Documents\Arduino\libraries\FastLED\src/platforms/esp/32/clockless_i2s_esp32.h:174: undefined reference to `i2s_is_initialized()'
c:\Users\....\Documents\Arduino\libraries\FastLED\src/platforms/esp/32/clockless_i2s_esp32.h:175: undefined reference to `i2s_define_bit_patterns(int, int, int)'
c:\Users\....\Documents\Arduino\libraries\FastLED\src/platforms/esp/32/clockless_i2s_esp32.h:176: undefined reference to `i2s_init(int)'
c:\Users\....\Documents\Arduino\libraries\FastLED\src/platforms/esp/32/clockless_i2s_esp32.h:177: undefined reference to `i2s_set_fill_buffer_callback(void (*)())'
c:\Users\....\Documents\Arduino\libraries\FastLED\src/platforms/esp/32/clockless_i2s_esp32.h:191: undefined reference to `i2s_setup_pin(int, int)'
C:\Users\Tester\AppData\Local\arduino\sketches\4EB0E45605664DFC0D280DCBCBFD79BC\sketch\Blink.ino.cpp.o: In function `ClocklessController<16, 60, 150, 90, (EOrder)66, 0, false, 5>::showPixels(PixelController<(EOrder)66, 1, 4294967295u>&)':
c:\Users\....\Documents\Arduino\libraries\FastLED\src/platforms/esp/32/clockless_i2s_esp32.h:258: undefined reference to `i2s_begin()'
C:\Users\Tester\AppData\Local\arduino\sketches\4EB0E45605664DFC0D280DCBCBFD79BC\sketch\Blink.ino.cpp.o: In function `ClocklessController<16, 60, 150, 90, (EOrder)66, 0, false, 5>::empty(unsigned int*)':
c:\Users\....\Documents\Arduino\libraries\FastLED\src/platforms/esp/32/clockless_i2s_esp32.h:202: undefined reference to `i2s_clear_dma_buffer(unsigned int*)'
c:\Users\....\Documents\Arduino\libraries\FastLED\src/platforms/esp/32/clockless_i2s_esp32.h:202: undefined reference to `i2s_clear_dma_buffer(unsigned int*)'
C:\Users\Tester\AppData\Local\arduino\sketches\4EB0E45605664DFC0D280DCBCBFD79BC\sketch\Blink.ino.cpp.o: In function `CMinWait<50>::wait()':
c:\Users\....\Documents\Arduino\libraries\FastLED\src/fastled_delay.h:29: undefined reference to `i2s_start()'
c:\Users\....\Documents\Arduino\libraries\FastLED\src/fastled_delay.h:29: undefined reference to `i2s_wait()'
C:\Users\Tester\AppData\Local\arduino\sketches\4EB0E45605664DFC0D280DCBCBFD79BC\sketch\Blink.ino.cpp.o: In function `ClocklessController<16, 60, 150, 90, (EOrder)66, 0, false, 5>::showPixels(PixelController<(EOrder)66, 1, 4294967295u>&)':
c:\Users\....\Documents\Arduino\libraries\FastLED\src/platforms/esp/32/clockless_i2s_esp32.h:291: undefined reference to `i2s_stop()'
collect2.exe: error: ld returned 1 exit status

Using library FastLED at version 3.9.7 in folder: C:\Users\....\Documents\Arduino\libraries\FastLED 
exit status 1

Compilation error: exit status 1
```

**Comment by zackees** [comment]
Fixed, will go into the next release.

For a work around, just  use `#include platforms/esp/32/i2s.cpp` directly in your sketch. On the next release, remove this line.

You should be using PlatformIO instead of Arduino, since it has build defines. It's backwards compatible too, as in your sketch can be compiled by both once it's converted into a platformio sketch.

https://github.com/FastLED/PlatformIO-Starter
